Kudu.Core/Jobs/DnxScriptHost.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Kudu.Core.Jobs
+{
+    class DnxScriptHost : ScriptHostBase
+    {
+        private static readonly string[] Supported = { "project.json" };
+
+        public DnxScriptHost()
+            : base("cmd", "/c runDnxWebJob.cmd")
+        {
+        }
+
+        public override IEnumerable<string> SupportedFileNames
```

Comment:
Now we have 2 paths - SupportedExtensions and SupportedFileNames. I wonder if a cleaner design would be to instead have an IScriptHost.GetEntryScript(FileInfoBase[] files) method that allows each host to control the scan logic itself, and return the entry script if found?